src/hclib-fptr-list.c

```diff
@@ -40,7 +40,11 @@ void hclib_register_func(hclib_fptr_list_t **list, int index, void *fptr,
 
 void *hclib_get_func_for(hclib_fptr_list_t *list, int index) {
     assert(list);
-    return (list->fptrs)[index];
+    if (index < list->capacity) {
```

Comment:
I think we should keep this behavior. Because these lists are lazily expanded, it is possible to get an index that is semantically valid but which the list hasn't been expanded to include yet (in which case the proper behavior would be to return the dummy value, NULL).